REST/GraphiQL Client

vwital opened the pull request:
1. Task:  https://github.com/rolling-scopes-school/tasks/blob/master/react/modules/tasks/final.md
2. Demo:
[Video demo](https://youtu.be/U7HwzInHGfI)
Screenshots 
![image](https://github.com/user-attachments/assets/7c321d65-16a9-4bc4-8de8-15bacc35af0a)
![image](https://github.com/user-attachments/assets/0b17077f-9393-4b05-b665-d47b23e1420d)
![image](https://github.com/user-attachments/assets/94a88b3e-592a-477f-945b-cd5933265daf)
![image](https://github.com/user-attachments/assets/56e1e6d8-9b96-4353-9c20-a4d5c11ff108)

4. Deployment : https://dev--rest-graphiql-app.netlify.app
5. Done: 16.09.24 02:00 /  Deadline: 16.09.24 03:00
6. Score 385/400
Main route - max 50 points
Sign In / Sign Up - max 50 points
RESTfull client - max 105 points ( Variables section that can shown or hidden, specified variables are included in the body. - 15 points not completed)
GraphiQL route - max 80 points
History route - max 50 points
General requirements - max 50 points

panakir committed:
feat: add base project setup (Tue, Aug 20, 2024 16:32)

panakir committed:
feat: add Header component, add favicon and title,  delete unused  main.tsx file (Tue, Aug 20, 2024 18:21)

panakir committed:
RC-5 Implment header component, RC-6 Implement footer component (#2)

* style: add normalize.css

* RC-5 feat: implement header

* RC-6 implement footer component, fix layout component

* style: add styles

panakir committed:
RC-7 feat: implement base routing (#1)

Dzehil02 committed:
RC-8 (Configuration and setup i18n) (#3)

* RC-8 feat: setup i18n next-intl

* feat: added translations

* feat: add navigation with ru-en prefix
add eslint rule to control correct Link import

* RC-8 feat: add locale switch button

* RC-8 feat: added descriptions for translations

vwital committed:
RC-10 (Welcome page layout) (#4)

* docs: update README (Mon, Aug 19, 2024 22:35)

* RC-10 feat: add fonts and fonts constats

* RC-10 feat: add responsive slider adaptive for main page

* RC-10 fix: adaptive layout

* RC-10 fix: adaptive layout

* RC-10 fix: adaptive cource link

* RC-10 fix: add inner fonts to assets

---------

Co-authored-by: panakir <panakir.webdev@gmail.com>

panakir committed:
RC-11 Implement static Rest Client Page layout (#5)

* fix: fix header sticky flicker bug

* RC-11 feat: implement static layout Rect Client Page

* refactor: replace routes folders to [locale]

* RC-11 feat: implement Rest client page  localization

vwital committed:
RC-12 feat: implement graphi page layout (#6)

panakir committed:
RC-16 Code refactor (#7)

* fix: add miss comma and brackets in message files

* refactor: add separate component Welcome page

* style[header]: add base styles for class selectors, add class selectors

* style[authorSlider]: fix naming css selectors, add transition placeholder

* chore[stylelint]: add camelCase pattern for selector class name

* refactor: replace locale switcher component im separate folder

* fix[welcome]: define width property with clamp function for .welcome__text

andrej-a reviewed (changes requested):
есть поводы подумать над файловой структурой
создавая новый компонент, отталкивайтесь от того, **где** он будет использоваться
не засоряйте общее рабочее пространство

не очень понравился касающийся стилей момент: слишком много указаний размеров в px, как будто верстка не резиновая совсем
хотелось бы, чтобы адаптив ваших компонентов вы переложили с себя на браузер, хотя бы отчасти

не увидел функционала
даже регистрации
ничего
возможно, я что-то сделал не так, но на скрине ниже все, что я вижу по урлам логина или авторизации
![image](https://github.com/user-attachments/assets/640eca89-c243-4c6a-8740-f62d24c7e3e1)


так же, судя по коду, у вас не то что бы готова хорошая такая часть приложения
до дедлайна 12 дней, звучит будто нужно поторопиться

Dzehil02 committed:
RC-9 Implement Login / Register page with client side validation (#9)

* RC-9 feat: add firebase, add Form

* RC-9 fix: moved graphi, history, rest to locale

* RC-9 feat: add styles for sign pages and Form

* RC-9 refactor: Changed styles of Form, sign pages
Add username to Form. Changed stylenames.

* RC-9 feat: add auth and redirect

* RC-9 feat: add logout

* RC-9 feat: add react-hook-form and yup to validate
Separate login and reg forms. Add schema

* RC-9 feat: add styles to _constants

* RC-9 feat: add main page button to header
add links sign-in/up on main page

* RC-9 feat: create useAuth hook, add hook to Header

* RC-9 feat: add sign-in/up route protection
add redirect to main page for logged user

* RC-9 feat: add token expires checking, and logout
add TOKEN_EXPIRATION 10 min; add Link to main page

* RC-9 fix: token experation time, logout

* RC-9 feat: add changes buttons after sign in/up
refactor welcome page

* RC-9 feat: add translations of errors of forms

* RC-9 fix: add middleware to check user auth
fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

* RC-9 fix: remove "use server" from Welcome page

* RC-9 refactor: add consts, change some styles

* RC-9 feat: protect routes rest, graphi, history
if user is not authorized

* RC-9 refactor: remove duplicate code.
change button styles, add constants, refactor code.

Dzehil02 committed:
RC-19 Implement not-found page, Error boundary, loading (#10)

* RC-9 feat: add firebase, add Form

* RC-9 fix: moved graphi, history, rest to locale

* RC-9 feat: add styles for sign pages and Form

* RC-9 refactor: Changed styles of Form, sign pages
Add username to Form. Changed stylenames.

* RC-9 feat: add auth and redirect

* RC-9 feat: add logout

* RC-9 feat: add react-hook-form and yup to validate
Separate login and reg forms. Add schema

* RC-9 feat: add styles to _constants

* RC-9 feat: add main page button to header
add links sign-in/up on main page

* RC-9 feat: create useAuth hook, add hook to Header

* RC-9 feat: add sign-in/up route protection
add redirect to main page for logged user

* RC-9 feat: add token expires checking, and logout
add TOKEN_EXPIRATION 10 min; add Link to main page

* RC-9 fix: token experation time, logout

* RC-9 feat: add changes buttons after sign in/up
refactor welcome page

* RC-9 feat: add translations of errors of forms

* RC-9 fix: add middleware to check user auth
fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

* RC-9 fix: remove "use server" from Welcome page

* RC-9 refactor: add consts, change some styles

* RC-9 feat: protect routes rest, graphi, history
if user is not authorized

* RC-9 refactor: remove duplicate code.
change button styles, add constants, refactor code.

* RC-19 feat: add not-found to app and to locale.

* RC-19 feat: add error.tsx

* RC-19 feat: add loading

* RC-19 fix: add layout to app, which is required

* RC-19 refactor: changed styles for WelcomePage

panakir committed:
RC -14 Implement the functionality of the REST client page (#11)

* fix: fix header sticky flicker bug

* RC-11 feat: implement static layout Rect Client Page

* refactor: replace routes folders to [locale]

* RC-11 feat: implement Rest client page  localization

* RC-14 feat: implement dynamic fields (headers and body fields) in form

* RC-14 feat: implement function to convert base64

* RC-16 refactor: replace pages folder into components folder

* RC-14 feat: add dynamic routes

* RC-14 refactor: refactor convertBase64 functions

* RC-14 feat: implement application url change when selecting method and url

* RC-14 styles: add error class selector, add color-error variable

* RC-14 feat: implement body field in RestForm, add buttons for add, delete, pretty body field, implement change application url when body field focus out

* RC-14 refactor: replace restForm and restResponse components into restClient/components

* RC-14 feat: add translator for buttons

* RC-14 refactor: add constants file, separate variable

* RC-14 feat: implement body buttons functionality,  refactor  method field

* RCC-14 feat: add translator for  headers and body fields

* RC-14 feat: implement add header key/value to url query

* RC-14 fix: fix body fields naming and default value

* RC-14 feat: add  and setup redux

* RC-14 feat: implement base server actions request and display response component

* refactor: delete unused files, delete comment in StoreProvider

* RC-14 refactor: replace slice file in app/lib/features folder

* fix: fix StoreProvider boundary

* chore: update package-lock file

* RC-14 feat: add check request for response display

* RC-14 feat: add submitButton  component, add translation for it

* RC-14 refactor: separate json viewer into a specific  component

* RC-14 feat: update method change (display body field according to method), implement  isDisabledAddBodyButton function

* RC-14 fix: fix bugs in body field, setup disabling body buttons, separate the function handleDefaultValue  into a specific file

* RC-14 style: add form adaptive style

* refactor: rename style file for json viewer component

* RC-14 refactor: change body wrapper  className

* RC-14 feat: add body request to server action

* fix: fix lint and format scripts

---------

Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

Dzehil02 committed:
RC-18 Implement testing of the components (#13)

* RC-9 feat: add firebase, add Form

* RC-9 fix: moved graphi, history, rest to locale

* RC-9 feat: add styles for sign pages and Form

* RC-9 refactor: Changed styles of Form, sign pages
Add username to Form. Changed stylenames.

* RC-9 feat: add auth and redirect

* RC-9 feat: add logout

* RC-9 feat: add react-hook-form and yup to validate
Separate login and reg forms. Add schema

* RC-9 feat: add styles to _constants

* RC-9 feat: add main page button to header
add links sign-in/up on main page

* RC-9 feat: create useAuth hook, add hook to Header

* RC-9 feat: add sign-in/up route protection
add redirect to main page for logged user

* RC-9 feat: add token expires checking, and logout
add TOKEN_EXPIRATION 10 min; add Link to main page

* RC-9 fix: token experation time, logout

* RC-9 feat: add changes buttons after sign in/up
refactor welcome page

* RC-9 feat: add translations of errors of forms

* RC-9 fix: add middleware to check user auth
fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

* RC-9 fix: remove "use server" from Welcome page

* RC-9 refactor: add consts, change some styles

* RC-9 feat: protect routes rest, graphi, history
if user is not authorized

* RC-9 refactor: remove duplicate code.
change button styles, add constants, refactor code.

* RC-19 feat: add not-found to app and to locale.

* RC-19 feat: add error.tsx

* RC-19 feat: add loading

* RC-19 fix: add layout to app, which is required

* RC-19 refactor: changed styles for WelcomePage

* RC-18 fix: translations and button style

* RC-18 feat: add test cases; coverage 40%

* RC-18 feat: add tests; cover 65%
off eslint rule @typescript-eslint/explicit-function-return-type for tests folder

* RC-18 feat: add tests; cover 85%

---------

Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

vwital committed:
Squashed commit of the following:

commit aacdf9093c59f7f348fdf9071ad7109231fd1a67
Author: Vital <wital128@tut.by>
Date:   Fri Sep 13 21:16:58 2024 +0300

    RC-15. fix: turn off pre-push

commit b2115787f79fe517bb6cdb0de5ee1418f7da6b08
Author: Vital <wital128@tut.by>
Date:   Fri Sep 13 20:44:01 2024 +0300

    RC-15. fix: remove unnecessary code

commit 924edbbc41e1c93d9b6e4565d48290476e7e5d30
Author: Vital <wital128@tut.by>
Date:   Fri Sep 13 20:37:00 2024 +0300

    Squashed commit of the following:

    commit f60a5b69031edb308705913eb15cdeca3901a9d7
    Author: Dzehil02 <73360230+Dzehil02@users.noreply.github.com>
    Date:   Fri Sep 13 19:01:22 2024 +0200

        RC-18 Implement testing of the components (#13)

        * RC-9 feat: add firebase, add Form

        * RC-9 fix: moved graphi, history, rest to locale

        * RC-9 feat: add styles for sign pages and Form

        * RC-9 refactor: Changed styles of Form, sign pages
        Add username to Form. Changed stylenames.

        * RC-9 feat: add auth and redirect

        * RC-9 feat: add logout

        * RC-9 feat: add react-hook-form and yup to validate
        Separate login and reg forms. Add schema

        * RC-9 feat: add styles to _constants

        * RC-9 feat: add main page button to header
        add links sign-in/up on main page

        * RC-9 feat: create useAuth hook, add hook to Header

        * RC-9 feat: add sign-in/up route protection
        add redirect to main page for logged user

        * RC-9 feat: add token expires checking, and logout
        add TOKEN_EXPIRATION 10 min; add Link to main page

        * RC-9 fix: token experation time, logout

        * RC-9 feat: add changes buttons after sign in/up
        refactor welcome page

        * RC-9 feat: add translations of errors of forms

        * RC-9 fix: add middleware to check user auth
        fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

        * RC-9 fix: remove "use server" from Welcome page

        * RC-9 refactor: add consts, change some styles

        * RC-9 feat: protect routes rest, graphi, history
        if user is not authorized

        * RC-9 refactor: remove duplicate code.
        change button styles, add constants, refactor code.

        * RC-19 feat: add not-found to app and to locale.

        * RC-19 feat: add error.tsx

        * RC-19 feat: add loading

        * RC-19 fix: add layout to app, which is required

        * RC-19 refactor: changed styles for WelcomePage

        * RC-18 fix: translations and button style

        * RC-18 feat: add test cases; coverage 40%

        * RC-18 feat: add tests; cover 65%
        off eslint rule @typescript-eslint/explicit-function-return-type for tests folder

        * RC-18 feat: add tests; cover 85%

        ---------

        Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

    commit 8c64edf4c094a052c5007d4bdf32fa95985b4e83
    Author: Kiryl Panamarou <70377995+panakir@users.noreply.github.com>
    Date:   Fri Sep 13 19:42:30 2024 +0300

        RC -14 Implement the functionality of the REST client page (#11)

        * fix: fix header sticky flicker bug

        * RC-11 feat: implement static layout Rect Client Page

        * refactor: replace routes folders to [locale]

        * RC-11 feat: implement Rest client page  localization

        * RC-14 feat: implement dynamic fields (headers and body fields) in form

        * RC-14 feat: implement function to convert base64

        * RC-16 refactor: replace pages folder into components folder

        * RC-14 feat: add dynamic routes

        * RC-14 refactor: refactor convertBase64 functions

        * RC-14 feat: implement application url change when selecting method and url

        * RC-14 styles: add error class selector, add color-error variable

        * RC-14 feat: implement body field in RestForm, add buttons for add, delete, pretty body field, implement change application url when body field focus out

        * RC-14 refactor: replace restForm and restResponse components into restClient/components

        * RC-14 feat: add translator for buttons

        * RC-14 refactor: add constants file, separate variable

        * RC-14 feat: implement body buttons functionality,  refactor  method field

        * RCC-14 feat: add translator for  headers and body fields

        * RC-14 feat: implement add header key/value to url query

        * RC-14 fix: fix body fields naming and default value

        * RC-14 feat: add  and setup redux

        * RC-14 feat: implement base server actions request and display response component

        * refactor: delete unused files, delete comment in StoreProvider

        * RC-14 refactor: replace slice file in app/lib/features folder

        * fix: fix StoreProvider boundary

        * chore: update package-lock file

        * RC-14 feat: add check request for response display

        * RC-14 feat: add submitButton  component, add translation for it

        * RC-14 refactor: separate json viewer into a specific  component

        * RC-14 feat: update method change (display body field according to method), implement  isDisabledAddBodyButton function

        * RC-14 fix: fix bugs in body field, setup disabling body buttons, separate the function handleDefaultValue  into a specific file

        * RC-14 style: add form adaptive style

        * refactor: rename style file for json viewer component

        * RC-14 refactor: change body wrapper  className

        * RC-14 feat: add body request to server action

        * fix: fix lint and format scripts

        ---------

        Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

    commit 6ba609c4b21854e0260fd571ab2076c6e455e41b
    Author: Dzehil02 <73360230+Dzehil02@users.noreply.github.com>
    Date:   Fri Sep 13 14:19:00 2024 +0200

        RC-19 Implement not-found page, Error boundary, loading (#10)

        * RC-9 feat: add firebase, add Form

        * RC-9 fix: moved graphi, history, rest to locale

        * RC-9 feat: add styles for sign pages and Form

        * RC-9 refactor: Changed styles of Form, sign pages
        Add username to Form. Changed stylenames.

        * RC-9 feat: add auth and redirect

        * RC-9 feat: add logout

        * RC-9 feat: add react-hook-form and yup to validate
        Separate login and reg forms. Add schema

        * RC-9 feat: add styles to _constants

        * RC-9 feat: add main page button to header
        add links sign-in/up on main page

        * RC-9 feat: create useAuth hook, add hook to Header

        * RC-9 feat: add sign-in/up route protection
        add redirect to main page for logged user

        * RC-9 feat: add token expires checking, and logout
        add TOKEN_EXPIRATION 10 min; add Link to main page

        * RC-9 fix: token experation time, logout

        * RC-9 feat: add changes buttons after sign in/up
        refactor welcome page

        * RC-9 feat: add translations of errors of forms

        * RC-9 fix: add middleware to check user auth
        fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

        * RC-9 fix: remove "use server" from Welcome page

        * RC-9 refactor: add consts, change some styles

        * RC-9 feat: protect routes rest, graphi, history
        if user is not authorized

        * RC-9 refactor: remove duplicate code.
        change button styles, add constants, refactor code.

        * RC-19 feat: add not-found to app and to locale.

        * RC-19 feat: add error.tsx

        * RC-19 feat: add loading

        * RC-19 fix: add layout to app, which is required

        * RC-19 refactor: changed styles for WelcomePage

    commit 5d269ec3ab63263d701aab3cde04251dbde13248
    Author: Dzehil02 <73360230+Dzehil02@users.noreply.github.com>
    Date:   Mon Sep 9 19:53:35 2024 +0200

        RC-9 Implement Login / Register page with client side validation (#9)

        * RC-9 feat: add firebase, add Form

        * RC-9 fix: moved graphi, history, rest to locale

        * RC-9 feat: add styles for sign pages and Form

        * RC-9 refactor: Changed styles of Form, sign pages
        Add username to Form. Changed stylenames.

        * RC-9 feat: add auth and redirect

        * RC-9 feat: add logout

        * RC-9 feat: add react-hook-form and yup to validate
        Separate login and reg forms. Add schema

        * RC-9 feat: add styles to _constants

        * RC-9 feat: add main page button to header
        add links sign-in/up on main page

        * RC-9 feat: create useAuth hook, add hook to Header

        * RC-9 feat: add sign-in/up route protection
        add redirect to main page for logged user

        * RC-9 feat: add token expires checking, and logout
        add TOKEN_EXPIRATION 10 min; add Link to main page

        * RC-9 fix: token experation time, logout

        * RC-9 feat: add changes buttons after sign in/up
        refactor welcome page

        * RC-9 feat: add translations of errors of forms

        * RC-9 fix: add middleware to check user auth
        fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

        * RC-9 fix: remove "use server" from Welcome page

        * RC-9 refactor: add consts, change some styles

        * RC-9 feat: protect routes rest, graphi, history
        if user is not authorized

        * RC-9 refactor: remove duplicate code.
        change button styles, add constants, refactor code.

commit 8f11addf51b359abc083e12e04606395b83985d6
Author: vwital <wital128@tut.by>
Date:   Fri Sep 13 13:49:17 2024 +0300

    RC-15. feat: add changing url

commit 2103ff545fa8098f5a5020d7bde6e93fe4a833cc
Author: vwital <wital128@tut.by>
Date:   Sun Sep 8 21:05:57 2024 +0300

    RC-15. refactor: move graphQL func to graphiForm file

commit 2630e72676d966b64d408c52b72fddf46eb10bef
Author: vwital <wital128@tut.by>
Date:   Sat Sep 7 21:03:26 2024 +0300

    RC-15 feat: add router utils

commit a62fdc7c518603c45d097283c7f2ee53a85234d7
Author: vwital <wital128@tut.by>
Date:   Fri Sep 6 21:28:26 2024 +0300

    RC-15. fix: headers adding

commit b2fbefd8a8832aeb4d42c59f376e31b41f9bf6d0
Author: vwital <wital128@tut.by>
Date:   Fri Sep 6 21:02:09 2024 +0300

    RC-15. feat: add documentation request

commit 30007d89a8572c40ca59545cdc5019d6d9d0ca68
Author: vwital <wital128@tut.by>
Date:   Fri Sep 6 07:22:50 2024 +0300

    RC-15. feat:add graphQL func

commit b4bc58958cba7ca2e53b84ba0e611551386d113d
Author: vwital <wital128@tut.by>
Date:   Tue Sep 3 22:57:41 2024 +0300

    RC-15 feat: divde graphi page into form and response fix: move pages to components folder

vwital committed:
RC-15. Imlemnt the functionality of GraphiQL page  (#14)

* RC-15 feat: divde graphi page into form and response fix: move pages to components folder

* RC-15. feat:add graphQL func

* RC-15. feat: add documentation request

* RC-15. fix: headers adding

* RC-15 feat: add router utils

* RC-15. refactor: move graphQL func to graphiForm file

* RC-15. feat: add changing url

* Squashed commit of the following:

commit f60a5b69031edb308705913eb15cdeca3901a9d7
Author: Dzehil02 <73360230+Dzehil02@users.noreply.github.com>
Date:   Fri Sep 13 19:01:22 2024 +0200

    RC-18 Implement testing of the components (#13)

    * RC-9 feat: add firebase, add Form

    * RC-9 fix: moved graphi, history, rest to locale

    * RC-9 feat: add styles for sign pages and Form

    * RC-9 refactor: Changed styles of Form, sign pages
    Add username to Form. Changed stylenames.

    * RC-9 feat: add auth and redirect

    * RC-9 feat: add logout

    * RC-9 feat: add react-hook-form and yup to validate
    Separate login and reg forms. Add schema

    * RC-9 feat: add styles to _constants

    * RC-9 feat: add main page button to header
    add links sign-in/up on main page

    * RC-9 feat: create useAuth hook, add hook to Header

    * RC-9 feat: add sign-in/up route protection
    add redirect to main page for logged user

    * RC-9 feat: add token expires checking, and logout
    add TOKEN_EXPIRATION 10 min; add Link to main page

    * RC-9 fix: token experation time, logout

    * RC-9 feat: add changes buttons after sign in/up
    refactor welcome page

    * RC-9 feat: add translations of errors of forms

    * RC-9 fix: add middleware to check user auth
    fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

    * RC-9 fix: remove "use server" from Welcome page

    * RC-9 refactor: add consts, change some styles

    * RC-9 feat: protect routes rest, graphi, history
    if user is not authorized

    * RC-9 refactor: remove duplicate code.
    change button styles, add constants, refactor code.

    * RC-19 feat: add not-found to app and to locale.

    * RC-19 feat: add error.tsx

    * RC-19 feat: add loading

    * RC-19 fix: add layout to app, which is required

    * RC-19 refactor: changed styles for WelcomePage

    * RC-18 fix: translations and button style

    * RC-18 feat: add test cases; coverage 40%

    * RC-18 feat: add tests; cover 65%
    off eslint rule @typescript-eslint/explicit-function-return-type for tests folder

    * RC-18 feat: add tests; cover 85%

    ---------

    Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

commit 8c64edf4c094a052c5007d4bdf32fa95985b4e83
Author: Kiryl Panamarou <70377995+panakir@users.noreply.github.com>
Date:   Fri Sep 13 19:42:30 2024 +0300

    RC -14 Implement the functionality of the REST client page (#11)

    * fix: fix header sticky flicker bug

    * RC-11 feat: implement static layout Rect Client Page

    * refactor: replace routes folders to [locale]

    * RC-11 feat: implement Rest client page  localization

    * RC-14 feat: implement dynamic fields (headers and body fields) in form

    * RC-14 feat: implement function to convert base64

    * RC-16 refactor: replace pages folder into components folder

    * RC-14 feat: add dynamic routes

    * RC-14 refactor: refactor convertBase64 functions

    * RC-14 feat: implement application url change when selecting method and url

    * RC-14 styles: add error class selector, add color-error variable

    * RC-14 feat: implement body field in RestForm, add buttons for add, delete, pretty body field, implement change application url when body field focus out

    * RC-14 refactor: replace restForm and restResponse components into restClient/components

    * RC-14 feat: add translator for buttons

    * RC-14 refactor: add constants file, separate variable

    * RC-14 feat: implement body buttons functionality,  refactor  method field

    * RCC-14 feat: add translator for  headers and body fields

    * RC-14 feat: implement add header key/value to url query

    * RC-14 fix: fix body fields naming and default value

    * RC-14 feat: add  and setup redux

    * RC-14 feat: implement base server actions request and display response component

    * refactor: delete unused files, delete comment in StoreProvider

    * RC-14 refactor: replace slice file in app/lib/features folder

    * fix: fix StoreProvider boundary

    * chore: update package-lock file

    * RC-14 feat: add check request for response display

    * RC-14 feat: add submitButton  component, add translation for it

    * RC-14 refactor: separate json viewer into a specific  component

    * RC-14 feat: update method change (display body field according to method), implement  isDisabledAddBodyButton function

    * RC-14 fix: fix bugs in body field, setup disabling body buttons, separate the function handleDefaultValue  into a specific file

    * RC-14 style: add form adaptive style

    * refactor: rename style file for json viewer component

    * RC-14 refactor: change body wrapper  className

    * RC-14 feat: add body request to server action

    * fix: fix lint and format scripts

    ---------

    Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

commit 6ba609c4b21854e0260fd571ab2076c6e455e41b
Author: Dzehil02 <73360230+Dzehil02@users.noreply.github.com>
Date:   Fri Sep 13 14:19:00 2024 +0200

    RC-19 Implement not-found page, Error boundary, loading (#10)

    * RC-9 feat: add firebase, add Form

    * RC-9 fix: moved graphi, history, rest to locale

    * RC-9 feat: add styles for sign pages and Form

    * RC-9 refactor: Changed styles of Form, sign pages
    Add username to Form. Changed stylenames.

    * RC-9 feat: add auth and redirect

    * RC-9 feat: add logout

    * RC-9 feat: add react-hook-form and yup to validate
    Separate login and reg forms. Add schema

    * RC-9 feat: add styles to _constants

    * RC-9 feat: add main page button to header
    add links sign-in/up on main page

    * RC-9 feat: create useAuth hook, add hook to Header

    * RC-9 feat: add sign-in/up route protection
    add redirect to main page for logged user

    * RC-9 feat: add token expires checking, and logout
    add TOKEN_EXPIRATION 10 min; add Link to main page

    * RC-9 fix: token experation time, logout

    * RC-9 feat: add changes buttons after sign in/up
    refactor welcome page

    * RC-9 feat: add translations of errors of forms

    * RC-9 fix: add middleware to check user auth
    fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

    * RC-9 fix: remove "use server" from Welcome page

    * RC-9 refactor: add consts, change some styles

    * RC-9 feat: protect routes rest, graphi, history
    if user is not authorized

    * RC-9 refactor: remove duplicate code.
    change button styles, add constants, refactor code.

    * RC-19 feat: add not-found to app and to locale.

    * RC-19 feat: add error.tsx

    * RC-19 feat: add loading

    * RC-19 fix: add layout to app, which is required

    * RC-19 refactor: changed styles for WelcomePage

commit 5d269ec3ab63263d701aab3cde04251dbde13248
Author: Dzehil02 <73360230+Dzehil02@users.noreply.github.com>
Date:   Mon Sep 9 19:53:35 2024 +0200

    RC-9 Implement Login / Register page with client side validation (#9)

    * RC-9 feat: add firebase, add Form

    * RC-9 fix: moved graphi, history, rest to locale

    * RC-9 feat: add styles for sign pages and Form

    * RC-9 refactor: Changed styles of Form, sign pages
    Add username to Form. Changed stylenames.

    * RC-9 feat: add auth and redirect

    * RC-9 feat: add logout

    * RC-9 feat: add react-hook-form and yup to validate
    Separate login and reg forms. Add schema

    * RC-9 feat: add styles to _constants

    * RC-9 feat: add main page button to header
    add links sign-in/up on main page

    * RC-9 feat: create useAuth hook, add hook to Header

    * RC-9 feat: add sign-in/up route protection
    add redirect to main page for logged user

    * RC-9 feat: add token expires checking, and logout
    add TOKEN_EXPIRATION 10 min; add Link to main page

    * RC-9 fix: token experation time, logout

    * RC-9 feat: add changes buttons after sign in/up
    refactor welcome page

    * RC-9 feat: add translations of errors of forms

    * RC-9 fix: add middleware to check user auth
    fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

    * RC-9 fix: remove "use server" from Welcome page

    * RC-9 refactor: add consts, change some styles

    * RC-9 feat: protect routes rest, graphi, history
    if user is not authorized

    * RC-9 refactor: remove duplicate code.
    change button styles, add constants, refactor code.

* RC-15. fix: remove unnecessary code

* RC-15. fix: turn off pre-push

vwital committed:
RC-15. fix: turn on tests

panakir committed:
RC-14 Implement the functionality of the REST client page (#15)

* fix: fix header sticky flicker bug

* RC-11 feat: implement static layout Rect Client Page

* refactor: replace routes folders to [locale]

* RC-11 feat: implement Rest client page  localization

* RC-14 feat: implement dynamic fields (headers and body fields) in form

* RC-14 feat: implement function to convert base64

* RC-16 refactor: replace pages folder into components folder

* RC-14 feat: add dynamic routes

* RC-14 refactor: refactor convertBase64 functions

* RC-14 feat: implement application url change when selecting method and url

* RC-14 styles: add error class selector, add color-error variable

* RC-14 feat: implement body field in RestForm, add buttons for add, delete, pretty body field, implement change application url when body field focus out

* RC-14 refactor: replace restForm and restResponse components into restClient/components

* RC-14 feat: add translator for buttons

* RC-14 refactor: add constants file, separate variable

* RC-14 feat: implement body buttons functionality,  refactor  method field

* RCC-14 feat: add translator for  headers and body fields

* RC-14 feat: implement add header key/value to url query

* RC-14 fix: fix body fields naming and default value

* RC-14 feat: add  and setup redux

* RC-14 feat: implement base server actions request and display response component

* refactor: delete unused files, delete comment in StoreProvider

* RC-14 refactor: replace slice file in app/lib/features folder

* fix: fix StoreProvider boundary

* chore: update package-lock file

* RC-14 feat: add check request for response display

* RC-14 feat: add submitButton  component, add translation for it

* RC-14 refactor: separate json viewer into a specific  component

* RC-14 feat: update method change (display body field according to method), implement  isDisabledAddBodyButton function

* RC-14 fix: fix bugs in body field, setup disabling body buttons, separate the function handleDefaultValue  into a specific file

* RC-14 style: add form adaptive style

* refactor: rename style file for json viewer component

* RC-14 refactor: change body wrapper  className

* RC-14 feat: add body request to server action

* fix: fix lint and format scripts

* RC-14 feat: implement add headers to request

* refactor: delete unused files

* RC-14 fix: bugfix window not defined issue

* RC-14 feat: implement removing header

* RC-14 feat: remove delete header button

* RC-14 fix: fix error during HEAD and OPTIONS methods

---------

Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

panakir committed:
fix: fix welcome page (buttons after sign-in)

vwital committed:
RC-15. fix: url change

vwital committed:
RC-15. feat: save data to local storage

vwital committed:
RC-15. fix: turn off tests

vwital committed:
RC-15. refactor: remove unused code

panakir committed:
RC-15 refactor: delete unused files

panakir committed:
chore: upd package-lock

Dzehil02 committed:
fix: Uncommented GraphiResponse to build project

Dzehil02 committed:
fix: remove tests for Graphi and Rest

panakir committed:
RC-15 refactor: rename folder

vwital committed:
Merge branch 'develop' of https://github.com/vwital/graphiql-app into develop

vwital committed:
RC-15. refacor: remove unused code

vwital committed:
RC-15. fix: turn on tests

panakir committed:
fix[welcomePage]: fix graphClient link href

panakir committed:
RC-20 Implement History Page (#17)

* RC-9 feat: add firebase, add Form

* RC-9 fix: moved graphi, history, rest to locale

* RC-9 feat: add styles for sign pages and Form

* fix: fix header sticky flicker bug

* RC-11 feat: implement static layout Rect Client Page

* RC-12 feat: implement graphi page layout

* RC-9 refactor: Changed styles of Form, sign pages
Add username to Form. Changed stylenames.

* RC-9 feat: add auth and redirect

* RC-9 feat: add logout

* refactor: replace routes folders to [locale]

* RC-11 feat: implement Rest client page  localization

* RC-9 feat: add react-hook-form and yup to validate
Separate login and reg forms. Add schema

* RC-9 feat: add styles to _constants

* RC-9 feat: add main page button to header
add links sign-in/up on main page

* RC-14 feat: implement dynamic fields (headers and body fields) in form

* RC-14 feat: implement function to convert base64

* RC-9 feat: create useAuth hook, add hook to Header

* RC-9 feat: add sign-in/up route protection
add redirect to main page for logged user

* RC-16 refactor: replace pages folder into components folder

* RC-14 feat: add dynamic routes

* RC-14 refactor: refactor convertBase64 functions

* RC-14 feat: implement application url change when selecting method and url

* RC-9 feat: add token expires checking, and logout
add TOKEN_EXPIRATION 10 min; add Link to main page

* RC-9 fix: token experation time, logout

* RC-9 feat: add changes buttons after sign in/up
refactor welcome page

* RC-9 feat: add translations of errors of forms

* RC-14 styles: add error class selector, add color-error variable

* RC-14 feat: implement body field in RestForm, add buttons for add, delete, pretty body field, implement change application url when body field focus out

* RC-9 fix: add middleware to check user auth
fix: off exlint rule require-await; add auth-actions for saving cookies; remove dispatch from forms;

* RC-9 fix: remove "use server" from Welcome page

* RC-9 refactor: add consts, change some styles

* RC-14 refactor: replace restForm and restResponse components into restClient/components

* RC-9 feat: protect routes rest, graphi, history
if user is not authorized

* RC-14 feat: add translator for buttons

* RC-14 refactor: add constants file, separate variable

* RC-14 feat: implement body buttons functionality,  refactor  method field

* RCC-14 feat: add translator for  headers and body fields

* RC-14 feat: implement add header key/value to url query

* RC-14 fix: fix body fields naming and default value

* RC-9 refactor: remove duplicate code.
change button styles, add constants, refactor code.

* RC-19 feat: add not-found to app and to locale.

* RC-19 feat: add error.tsx

* RC-19 feat: add loading

* RC-19 fix: add layout to app, which is required

* RC-14 feat: add  and setup redux

* RC-14 feat: implement base server actions request and display response component

* refactor: delete unused files, delete comment in StoreProvider

* RC-19 refactor: changed styles for WelcomePage

* RC-18 fix: translations and button style

* RC-18 feat: add test cases; coverage 40%

* RC-14 refactor: replace slice file in app/lib/features folder

* fix: fix StoreProvider boundary

* chore: update package-lock file

* RC-14 feat: add check request for response display

* RC-18 feat: add tests; cover 65%
off eslint rule @typescript-eslint/explicit-function-return-type for tests folder

* RC-18 feat: add tests; cover 85%

* RC-14 feat: add submitButton  component, add translation for it

* RC-14 refactor: separate json viewer into a specific  component

* RC-14 feat: update method change (display body field according to method), implement  isDisabledAddBodyButton function

* RC-14 fix: fix bugs in body field, setup disabling body buttons, separate the function handleDefaultValue  into a specific file

* RC-14 style: add form adaptive style

* refactor: rename style file for json viewer component

* RC-14 refactor: change body wrapper  className

* RC-14 feat: add body request to server action

* fix: fix lint and format scripts

* RC-20 fix: translations, test, del Providers.

* RC-20 fix: translations and WelcomePage

* RC-20 fix: test for graphi page

* RC-20 add base components to History page, add styles to EmptyHistory component

* RC-20 refactor: moved auth-actions to app/actions

* RC-20 feat: add tests, coverage =80%

* RC-20 feat: implement history for rest client, add styles

* RC-20 feat: add translation into History page

---------

Co-authored-by: dzehil02 <dzehil02@gmail.com>
Co-authored-by: vwital <wital128@tut.by>
Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

panakir committed:
RC-15 Implement the functionality of the GraphiQL page (#18)

* fix: fix header sticky flicker bug

* RC-11 feat: implement static layout Rect Client Page

* refactor: replace routes folders to [locale]

* RC-11 feat: implement Rest client page  localization

* RC-14 feat: implement dynamic fields (headers and body fields) in form

* RC-14 feat: implement function to convert base64

* RC-16 refactor: replace pages folder into components folder

* RC-14 feat: add dynamic routes

* RC-14 refactor: refactor convertBase64 functions

* RC-14 feat: implement application url change when selecting method and url

* RC-14 styles: add error class selector, add color-error variable

* RC-14 feat: implement body field in RestForm, add buttons for add, delete, pretty body field, implement change application url when body field focus out

* RC-14 refactor: replace restForm and restResponse components into restClient/components

* RC-14 feat: add translator for buttons

* RC-14 refactor: add constants file, separate variable

* RC-14 feat: implement body buttons functionality,  refactor  method field

* RCC-14 feat: add translator for  headers and body fields

* RC-14 feat: implement add header key/value to url query

* RC-14 fix: fix body fields naming and default value

* RC-14 feat: add  and setup redux

* RC-14 feat: implement base server actions request and display response component

* refactor: delete unused files, delete comment in StoreProvider

* RC-14 refactor: replace slice file in app/lib/features folder

* fix: fix StoreProvider boundary

* chore: update package-lock file

* RC-14 feat: add check request for response display

* RC-14 feat: add submitButton  component, add translation for it

* RC-14 refactor: separate json viewer into a specific  component

* RC-14 feat: update method change (display body field according to method), implement  isDisabledAddBodyButton function

* RC-14 fix: fix bugs in body field, setup disabling body buttons, separate the function handleDefaultValue  into a specific file

* RC-14 style: add form adaptive style

* refactor: rename style file for json viewer component

* RC-14 refactor: change body wrapper  className

* RC-14 feat: add body request to server action

* fix: fix lint and format scripts

* RC-14 feat: implement add headers to request

* refactor: delete unused files

* RC-14 fix: bugfix window not defined issue

* RC-14 feat: implement removing header

* RC-14 feat: remove delete header button

* RC-14 fix: fix error during HEAD and OPTIONS methods

* RC-14 fix: fix header default value

* RC-14 fix: fix bugs during OPTIONS request

* RC-15 feat: implement url, sdl url and headers default values, add documentation viewer

* RC-15 feat: implement docs viewer, and styled it

* RC-15 fix: fix display default value request query

* RC-15 fix: fix display default value request query

* fix: fix conflict after merge

* Merge branch 'develop' into v/fix/graphi-page

---------

Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

panakir committed:
RC-14 Implement the functionality of the REST client page (#20)

* fix: fix header sticky flicker bug

* RC-11 feat: implement static layout Rect Client Page

* refactor: replace routes folders to [locale]

* RC-11 feat: implement Rest client page  localization

* RC-14 feat: implement dynamic fields (headers and body fields) in form

* RC-14 feat: implement function to convert base64

* RC-16 refactor: replace pages folder into components folder

* RC-14 feat: add dynamic routes

* RC-14 refactor: refactor convertBase64 functions

* RC-14 feat: implement application url change when selecting method and url

* RC-14 styles: add error class selector, add color-error variable

* RC-14 feat: implement body field in RestForm, add buttons for add, delete, pretty body field, implement change application url when body field focus out

* RC-14 refactor: replace restForm and restResponse components into restClient/components

* RC-14 feat: add translator for buttons

* RC-14 refactor: add constants file, separate variable

* RC-14 feat: implement body buttons functionality,  refactor  method field

* RCC-14 feat: add translator for  headers and body fields

* RC-14 feat: implement add header key/value to url query

* RC-14 fix: fix body fields naming and default value

* RC-14 feat: add  and setup redux

* RC-14 feat: implement base server actions request and display response component

* refactor: delete unused files, delete comment in StoreProvider

* RC-14 refactor: replace slice file in app/lib/features folder

* fix: fix StoreProvider boundary

* chore: update package-lock file

* RC-14 feat: add check request for response display

* RC-14 feat: add submitButton  component, add translation for it

* RC-14 refactor: separate json viewer into a specific  component

* RC-14 feat: update method change (display body field according to method), implement  isDisabledAddBodyButton function

* RC-14 fix: fix bugs in body field, setup disabling body buttons, separate the function handleDefaultValue  into a specific file

* RC-14 style: add form adaptive style

* refactor: rename style file for json viewer component

* RC-14 refactor: change body wrapper  className

* RC-14 feat: add body request to server action

* fix: fix lint and format scripts

* RC-14 feat: implement add headers to request

* refactor: delete unused files

* RC-14 fix: bugfix window not defined issue

* RC-14 feat: implement removing header

* RC-14 feat: remove delete header button

* RC-14 fix: fix error during HEAD and OPTIONS methods

* RC-14 fix: fix header default value

* RC-14 fix: fix bugs during OPTIONS request

---------

Co-authored-by: Vital <86975341+vwital@users.noreply.github.com>

Dzehil02 committed:
RC-18 Implement testing of the components (#19)

* RC-18 fix: welcomePage, restClientFormAction tests

* RC-18 feat: add tests; cover 72%

* RC-18 feat: add tests; cover 85%

Dzehil02 committed:
hotfix: graph test and jsonViewer import name

Dzehil02 committed:
hotfix: link graph; token exp = 30min
remove couple tests for graph. Cover 82%

panakir committed:
RC-16 Code refactor (#21)

* refactor: refactor app architecture

* refactor: refactor name class selectors

* refactor: refactor AuthorsSlider component

* refactor: refactor classNames notation

* refactor: refactor header component

* refactor: refactor auth forms

* refactor: refactor base styles

panakir committed:
fix: fix test and header links
